core/src/main/java/com/novoda/merlin/receiver/ConnectivityCallbacks.java

```diff
@@ -0,0 +1,50 @@
+package com.novoda.merlin.receiver;
+
+import android.annotation.TargetApi;
+import android.net.ConnectivityManager;
+import android.net.Network;
+import android.net.NetworkInfo;
+import android.os.Build;
+
+import com.novoda.merlin.service.MerlinService;
+
+@TargetApi(Build.VERSION_CODES.LOLLIPOP)
+public class ConnectivityCallbacks extends ConnectivityManager.NetworkCallback {
+
+    private final ConnectivityManager connectivityManager;
+    private final MerlinService merlinService;
```

Comment:
I am not 100% aligned with the project but it is generally not a good idea to hold on to Android Service. It can be killed at any time. Especially if it is not bound to the Activity in the foreground. 

Usually it is better practice to use `startService` with different Actions and they will call `onStartCommad` on an already initialized Service.